sdk/azidentity/workload_identity.go

```diff
@@ -139,3 +154,205 @@ func (w *WorkloadIdentityCredential) getAssertion(context.Context) (string, erro
+	}
+
+	transporter, err := newIdentityBindingTransport(
+		kubernetesCAFile, kubernetesSNIName, kubernetesTokenEndpointStr,
+		caco.Transport,
+	)
+	if err != nil {
+		return err
+	}
+	caco.Transport = transporter
+	return nil
+}
+
+const (
+	tokenEndpointSuffix = "/oauth2/v2.0/token"
+	caReloadInterval    = 10 * time.Minute
+)
+
+// identityBindingTransport is a custom HTTP transport that redirects token requests
+// to the Kubernetes token endpoint when in identity binding mode
+type identityBindingTransport struct {
+	caFile              string
+	sniName             string
+	tokenEndpoint       *url.URL
+	fallbackTransporter policy.Transporter
+
+	mtx *sync.RWMutex
+
+	nextRead  time.Time
+	currentCA []byte
+	transport *http.Transport
+}
+
+func newIdentityBindingTransport(
+	caFile, sniName, tokenEndpointStr string,
+	fallbackTransporter policy.Transporter,
+) (*identityBindingTransport, error) {
+	tokenEndpoint, err := url.Parse(tokenEndpointStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse token endpoint URL %q: %w", tokenEndpointStr, err)
+	}
+
+	if fallbackTransporter == nil {
+		// FIXME: can we callback to the defaultHTTPClient from azcore/runtime?
+		fallbackTransporter = http.DefaultClient
+	}
+
+	initialTransport := func() *http.Transport {
+		// try reusing the user provided transport if available
+		if httpClient, ok := fallbackTransporter.(*http.Client); ok {
+			if transport, ok := httpClient.Transport.(*http.Transport); ok {
+				return transport.Clone()
+			}
+		}
+
+		// if the user did not provide a policy.Transporter or it's not a *http.Client,
+		// we fall back to the default one.
+		// FIXME: can we callback to the defaultHTTPClient from azcore/runtime?
+		if transport, ok := http.DefaultTransport.(*http.Transport); ok {
+			return transport.Clone()
+		}
+
+		// this should not happen, but if the user mutates the net/http.DefaultTransport
+		// to something else, we fall back to a sane default
+		return &http.Transport{
+			ForceAttemptHTTP2:   true,
+			MaxIdleConns:        100,
+			IdleConnTimeout:     90 * time.Second,
+			TLSHandshakeTimeout: 10 * time.Second,
+		}
+	}()
+
+	tr := &identityBindingTransport{
+		caFile:              caFile,
+		sniName:             sniName,
+		tokenEndpoint:       tokenEndpoint,
+		fallbackTransporter: fallbackTransporter,
+		mtx:                 &sync.RWMutex{},
+		transport:           initialTransport,
+	}
+
+	// perform an initial load to surface any issues with the CA file and transport settings.
+	// Lock is not held here as this is called in the constructor
+	if err := tr.reloadCA(); err != nil {
+		return nil, err
+	}
+
+	return tr, nil
+}
+
+func (i *identityBindingTransport) Do(req *http.Request) (*http.Response, error) {
+	if !strings.HasSuffix(req.URL.Path, tokenEndpointSuffix) {
+		// not a token request, fallback to the original transporter
+		return i.fallbackTransporter.Do(req)
```

Comment:
I think the client does discovery and figures out the actual token endpoint at runtime, so to me it would be safer to check for that exact value instead of hard coding something like this.  Entra is free to change its token endpoint at any time.